src/Simulation/CSharpGeneration.Tests/SimulationCodeTests.fs

```diff
@@ -248,6 +248,53 @@ namespace N1
         List.zip expected actual
         |> List.iter Assert.Equal<'Z>
 
+    [<Fact>]
+    let ``testGeneratedFileNames`` () =
+        let compilation = parse [
+                Path.Combine("Circuits", "Intrinsic.qs")
+                Path.Combine("Circuits", "HelloWorld.qs")
+                Path.Combine("Circuits", "SubDirectory", "HelloWorld.qs")
+                Path.Combine("Circuits", "SubDirectory", "HelloOther.qs")
+                ]
+        let transformed = ref { Namespaces = ImmutableArray<QsNamespace>.Empty; EntryPoints = ImmutableArray<QsQualifiedName>.Empty }
+        
+        let rewriteStep = Emitter() :> IRewriteStep
+        
+        let outputDir = "output" |> Path.GetFullPath
+        let outputDirSrc = Path.Combine(outputDir, "src")
+
+        rewriteStep.AssemblyConstants.Add(AssemblyConstants.OutputPath, outputDir)
+        rewriteStep.Transformation(compilation, transformed) |> ignore
+
+        let intrinsic = Path.Combine(outputDirSrc, "Intrinsic.g.cs")
+        let helloWorld = Path.Combine(outputDirSrc, "HelloWorld.g.cs")
+        let helloWorld1 = Path.Combine(outputDirSrc, "HelloWorld.g1.cs")
+        let helloOther = Path.Combine(outputDirSrc, "HelloOther.g.cs")
+
+        let mutable allFilesFound = true
+        
+        let deleteFile filePath =
+            if File.Exists filePath then
+                File.Delete filePath
+            else
+                allFilesFound <- false
+
+        intrinsic |> deleteFile
```

Comment:
In general I'd recommend to delete files from disk at the start of the test rather than at the end for tests that check for files on disk. Not deleting it at the beginning relies somewhat on the tests being run on a clean machine; if these files for some reason already exists on disk then the first time the test is run it will succeed.

Reply:
I've updated the test to do both the check/delete before the transformation, and the check/delete after. You are right that the test should delete the files if they exist beforehand, otherwise it will compromise the test, but I also don't want to leave these files on the system after the test either, so I updated the test to do both.